addons/website_portal_sale/controllers/main.py

```diff
@@ -175,6 +175,19 @@ def portal_my_invoices(self, page=1, date_begin=None, date_end=None, **kw):
         })
         return request.render("website_portal_sale.portal_my_invoices", values)
 
+    @http.route(['/my/invoices/<int:invoice_id>'], type='http', auth="user", website=True)
+    def portal_get_invoice(self, invoice_id=None, **kw):
+        invoice = request.env['account.invoice'].browse([invoice_id])
+        try:
+            invoice.check_access_rights('read')
+            invoice.check_access_rule('read')
+        except AccessError:
+            return request.render("website.403")
```

Comment:
This will return a httpcode 200, which is not correct.
Doesn't letting the AccessError bubble up already handle the case?